common-items.lic

```diff
@@ -13,22 +13,23 @@ module DRCI
   def buy_item(room, item)
     DRCT.walk_to(room)
 
-    case DRC.bput("buy #{item}", /prepared to offer it to you for (.*) kronars/, /Let me but ask the humble sum of (.*) coins/, /for a mere (.*) kronars/, /I can let that go for...(.*) kronars/, /That'll cost you (.*) lirums/, 'You decide to purchase', 'Buy what', /I'll give that to you for (.*) dokoras/, /I ask that you give (.*) copper dokoras/)
-    when /prepared to offer it to you for (.*) kronars/
-      amount = Regexp.last_match(1)
-    when /Let me but ask the humble sum of (.*) coins/
-      amount = Regexp.last_match(1)
-    when /for a mere (.*) kronars/
-      amount = Regexp.last_match(1)
-    when /I can let that go for...(.*) kronars/
-      amount = Regexp.last_match(1)
-    when /That'll cost you (.*) lirums/
-      amount = Regexp.last_match(1)
-    when /I'll give that to you for (.*) dokoras/
-      amount = Regexp.last_match(1)
-    when /I ask that you give (.*) copper dokoras/
-      amount = Regexp.last_match(1)
+    # Amount to pay should be first capturing group.
+    patterns = [/prepared to offer it to you for (.*) (?:kronar|lirum|dokora)s?/,
+                /Let me but ask the humble sum of (.*) coins/,
+                /for a mere (.*) (?:kronar|lirum|dokora)s?/,
+                /I can let that go for...(.*) (?:kronar|lirum|dokora)s?/,
+                /That'll cost you (.*) (?:kronar|lirum|dokora)s?/,
+                'You decide to purchase',
+                'Buy what',
+                /I'll give that to you for (.*) (?:kronar|lirum|dokora)s?/,
+                /I ask that you give (.*) copper (?:kronar|lirum|dokora)s?/,
+                /it'll be (.*) (?:kronar|lirum|dokora)s?/]
+    
+    match = DRC.bput("buy #{item}", *patterns)
+    if match
+      patterns.each {|p| break if p.match(match)}
```

Comment:
Do you need to explicitly match against the patterns? That is, won't `Regexp.last_match(1)` already be the amount based on the match done within `bput`?

Reply:
I initially thought the same thing, but it turns out Regexp#last_match only retains the match within the current method scope.  I tried exactly that and the match is always nil.  Relevant docs:

https://ruby-doc.org/core-2.1.1/Regexp.html#method-c-last_match

Reply:
Good to know!